functions/form-payment-reconciliation/index.js

```diff
@@ -0,0 +1,291 @@
+/**
+ * Main file that runs payment reconciliation CRON job.
+ * Required env vars:
+ * - AWS_REGION: the aws region which the
+ * - SSM_ENV_SITE_NAME: ['prod', 'staging', 'staging-alt', 'staging-alt2', 'uat']
+ *
+ * Required parameters in parameter store `<SSM_ENV_SITE_NAME>-cron-payment`:
+ * - CRON_PAYMENT_API_SECRET: the shared secret to validate protected routes in FormSG API
+ * - SLACK_API_SECRET: the API key for posting reports to slack
+ */
+
+const { SSMClient, GetParameterCommand } = require('@aws-sdk/client-ssm')
+const { generateLinkToCloudwatch, getPostToSlack } = require('./src/utils')
+
+const AWS_REGION = process.env.AWS_REGION
+
+const ENV_SITE_NAME = process.env.SSM_ENV_SITE_NAME
+const CRON_API_PREFIX = `https://${
+  ENV_SITE_NAME === 'prod' ? '' : `${ENV_SITE_NAME}.`
+}form.gov.sg/api/v3/payments/reconcile/`
+
+const PARAMETER_STORE_NAME = `${ENV_SITE_NAME}-cron-payment`
+const CRON_PAYMENT_API_SECRET_KEY = 'CRON_PAYMENT_API_SECRET'
+const CRON_PAYMENT_SLACK_SECRET_KEY = 'SLACK_API_SECRET'
+
+const API_AUTH_HEADER = 'x-formsg-cron-payment-secret'
+
+const MAX_AGE_HRS_EVENTS = 7
```

Comment:
I extended it to 24 hours. 

The logic here is that if an event's re-processing failed, we should look into it. If we didn't look into it, it should continue to be retried and reported as failed in future CRON jobs. In the case that we looked into it but deemed it not to be a relevant event, we should let it exit the system at some point, so that we don't keep it around. I think a reasonable timeframe to assume we will get to looking at failed events is 24h (e.g. if this occurs while we are asleep, we need at least 8 hours buffer, assuming we all sleep at least 8 hours a night 🤡 ).